core/store/migrations/migration1587580235/migrate.go

```diff
@@ -1,27 +1,21 @@
 package migration1587580235
 
 import (
-	"time"
-
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/jinzhu/gorm"
-	"github.com/pkg/errors"
-	"github.com/smartcontractkit/chainlink/core/store/models"
 )
 
-type logConsumption struct {
-	ID           *models.ID  `gorm:"primary_key"`
-	BlockHash    common.Hash `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	ConsumerType string      `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	ConsumerID   *models.ID  `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	LogIndex     uint        `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	CreatedAt    time.Time
-}
-
-// Migrate adds the LogConsumption table
+// Migrate adds the LogConsumption table and adds a uniqueness constraint on the
+// combination of block_hash / log_index / consumer_id
 func Migrate(tx *gorm.DB) error {
-	if err := tx.AutoMigrate(&logConsumption{}).Error; err != nil {
-		return errors.Wrap(err, "could not add log_consumption table")
-	}
-	return nil
+	return tx.Exec(`
+	CREATE TABLE "log_consumptions" (
+		"id" serial primary key NOT NULL,
+		"block_hash" bytea NOT NULL,
+		"log_index" integer NOT NULL,
```

Comment:
It's not expected to be large, right?

Reply:
@samsondav Some back-of-the-napkin math to find an upper bound:

gas limit per block: ~10,000,000
gas cost per event emission per byte: 8

If an entire block was filled with txs that did nothing but emit one byte events, we'd end up with..

10,000,000 / 8 = 1,250,000 different log events.

Which is still a factor of 1,000 smaller than the max value for integers. So I think we're good here.